syntax = "proto3";

service Monitor {
  rpc NewTargetRequest (MonitorRequest) returns (MonitorReply) {}
  rpc DelTargetRequest (VnfId) returns (MonitorReply) {}
}

message MonitorRequest {
  string vnf_id = 1;
  repeated Vdu vdus = 2;
}

message VnfId {
  string vnf_id = 1;
}

message Vdu {
  string vdu_name = 1;
  string mgmt_ip = 2;
  string exporter_port = 3;
  repeated Alert alerts = 4;
}

message Alert {
  string group = 1;
  string alert = 2;
  string expr = 3;
  string duration = 4;
  map<string, string> labels = 5;
  map<string, string> annotations = 6;
}

message MonitorReply {
  string message = 1;
  int32 status = 2;
}
